File: cakk-api/src/main/java/com/cakk/api/controller/user/MyPageController.java
package com.cakk.api.controller.user;

import jakarta.validation.Valid;

import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;

import com.cakk.api.annotation.SignInUser;
import com.cakk.api.dto.request.like.HeartCakeSearchRequest;
import com.cakk.api.dto.request.like.HeartCakeShopSearchRequest;
import com.cakk.api.dto.request.user.ProfileUpdateRequest;
import com.cakk.api.dto.response.like.HeartCakeImageListResponse;
import com.cakk.api.dto.response.like.HeartCakeShopListResponse;
import com.cakk.api.dto.response.user.ProfileInformationResponse;
import com.cakk.api.service.like.HeartService;
import com.cakk.api.service.user.UserService;
import com.cakk.common.response.ApiResponse;
import com.cakk.domain.mysql.entity.user.User;

@RestController
@RequiredArgsConstructor
@RequestMapping("/me")
public class MyPageController {

	private final UserService userService;
	private final HeartService heartService;

	@GetMapping
	public ApiResponse<ProfileInformationResponse> profile(
		@SignInUser User user
	) {
		final ProfileInformationResponse response = userService.findProfile(user);

		return ApiResponse.success(response);
	}

	@PutMapping
	public ApiResponse<Void> modify(
		@SignInUser User user,
		@RequestBody @Valid ProfileUpdateRequest request
	) {
		userService.updateInformation(user, request);

		return ApiResponse.success();
	}

	@DeleteMapping
	public ApiResponse<Void> withdraw(
		@SignInUser User user
	) {
		userService.withdraw(user);

		return ApiResponse.success();
	}

	@GetMapping("/heart-cakes")
	public ApiResponse<HeartCakeImageListResponse> heartCakeList(
		@SignInUser User user,
		@Valid @ModelAttribute HeartCakeSearchRequest request
	) {
		final HeartCakeImageListResponse response = heartService.findCakeImagesByCursorAndHeart(request, user);

		return ApiResponse.success(response);
	}

	@GetMapping("/heart-shops")
	public ApiResponse<HeartCakeShopListResponse> heartCakeList(
		@SignInUser User user,
		@Valid @ModelAttribute HeartCakeShopSearchRequest request
	) {
		final HeartCakeShopListResponse response = heartService.findCakeShopByCursorAndHeart(request, user);

		return ApiResponse.success(response);
	}
}

File: cakk-api/src/main/java/com/cakk/api/controller/user/MyPageController.kt
package com.cakk.api.controller.user

import jakarta.validation.Valid

import org.springframework.web.bind.annotation.*

import com.cakk.api.annotation.SignInUser
import com.cakk.api.dto.request.like.HeartCakeSearchRequest
import com.cakk.api.dto.request.like.HeartCakeShopSearchRequest
import com.cakk.api.dto.request.user.ProfileUpdateRequest
import com.cakk.api.mapper.supplyHeartCakeSearchParamBy
import com.cakk.api.mapper.supplyHeartCakeShopSearchParamBy
import com.cakk.core.dto.response.like.HeartCakeImageListResponse
import com.cakk.core.dto.response.like.HeartCakeShopListResponse
import com.cakk.core.dto.response.user.ProfileInformationResponse
import com.cakk.api.mapper.supplyProfileUpdateParamBy
import com.cakk.core.service.like.HeartService
import com.cakk.core.service.user.UserService
import com.cakk.common.response.ApiResponse
import com.cakk.domain.mysql.entity.user.User

@RestController
@RequestMapping("/me")
class MyPageController(
    private val userService: UserService,
    private val heartService: HeartService
) {

    @GetMapping
    fun profile(
		@SignInUser user: User
    ): ApiResponse<ProfileInformationResponse> {
        val response = userService.findProfile(user)

		return ApiResponse.success(response)
    }

    @PutMapping
    fun modify(
		@SignInUser user: User,
		@RequestBody @Valid request: ProfileUpdateRequest
    ): ApiResponse<Unit> {
		val param = supplyProfileUpdateParamBy(request, user)
        userService.updateInformation(param)

		return ApiResponse.success()
    }

    @DeleteMapping
    fun withdraw(
		@SignInUser user: User
    ): ApiResponse<Unit> {
        userService.withdraw(user)

		return ApiResponse.success()
    }

    @GetMapping("/heart-cakes")
    fun heartCakeList(
		@SignInUser user: User,
		@ModelAttribute @Valid request: HeartCakeSearchRequest
    ): ApiResponse<HeartCakeImageListResponse> {
		val param = supplyHeartCakeSearchParamBy(request, user)
        val response = heartService.searchCakeImagesByCursorAndHeart(param)

		return ApiResponse.success(response)
    }

    @GetMapping("/heart-shops")
    fun heartCakeList(
		@SignInUser user: User,
		@ModelAttribute @Valid request: HeartCakeShopSearchRequest
    ): ApiResponse<HeartCakeShopListResponse> {
		val param = supplyHeartCakeShopSearchParamBy(request, user)
        val response = heartService.searchCakeShopByCursorAndHeart(param)

		return ApiResponse.success(response)
    }
}
